### PR TITLE
Add ProjectKeyboard Alice PCB to QMK

### PR DESCRIPTION
**Pull request by awkannan**
## Description

Add ProjectKeyboard Alice PCB to QMK
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
Looks like you have some conflicts here.

**Comment by awkannan** [comment]
@drashna Yup! It wasn't quite ready to be looked at. It's good now.

File: keyboards/projectkb/alice/alice.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "quantum.h"
+
+#define KNO KC_NO
+
+#define LAYOUT_default( \
```
Comment (mechmerlin) [inline keyboards/projectkb/alice/alice.h]: perhaps just call it `LAYOUT` as there really isn't any other layouts defined?

File: keyboards/projectkb/alice/readme.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# ProjectKeyboard Alice PCB
+
+ProjectKeyboard Alice Keyboard Replacement PCB
+
+Keyboard Maintainer: onefiftynine 
+
+Hardware Supported: STM32F072CBT6  
+
```
Comment (mechmerlin) [inline keyboards/projectkb/alice/readme.md]:
```suggestion
```

File: keyboards/projectkb/alice/halconf.h [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+ * @brief   HAL configuration header.
+ * @details HAL configuration file, this file allows to enable or disable the
+ *          various device drivers from your application. You may also use
+ *          this file in order to override the device drivers default settings.
+ *
+ * @addtogroup HAL_CONF
+ * @{
+ */
+
+#ifndef _HALCONF_H_
```
Comment (mechmerlin) [inline keyboards/projectkb/alice/halconf.h]:
```suggestion
#pragma once
```

File: keyboards/projectkb/alice/halconf.h [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+#define _HALCONF_H_
```
Comment (mechmerlin) [inline keyboards/projectkb/alice/halconf.h]:
```suggestion

```

File: keyboards/projectkb/alice/halconf.h [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+
+/**
+ * @brief   Enables synchronous APIs.
+ * @note    Disabling this option saves both code and data space.
+ */
+#if !defined(USB_USE_WAIT) || defined(__DOXYGEN__)
+#define USB_USE_WAIT                TRUE
+#endif
+
+#endif /* _HALCONF_H_ */
```
Comment (mechmerlin) [inline keyboards/projectkb/alice/halconf.h]:
```suggestion

```

File: keyboards/projectkb/alice/alice.c [hunk 6]
```diff
@@ -0,0 +1,27 @@
+#include "alice.h"
+
+void matrix_init_board(void){
+  palSetPadMode(GPIOA, 0, PAL_MODE_OUTPUT_PUSHPULL);
```
Comment (drashna) [inline keyboards/projectkb/alice/alice.c]: Would you mind converting these to the GPIO commands, if possible?
https://docs.qmk.fm/#/internals_gpio_control

https://github.com/qmk/qmk_firmware/blob/master/quantum/quantum.h#L166-L184

**Comment by mechmerlin** [comment]
@awkannan hey Upas, any traction on this, I don't want this to get lost in the rabbit hole.

File: keyboards/projectkb/alice/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+#define _FN1 1
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
+};
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT_default(
+    KC_PSCR, KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,   KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_DEL,  KC_BSPC, \
```
Comment (fauxpark) [inline keyboards/projectkb/alice/keymaps/default/keymap.c]: The backslashes at the end of these lines are not necessary. You can keep them if you want as it isn't hurting anything, but IMO it looks neater and less "hack-y".

File: keyboards/projectkb/alice/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+#define _BASE 0
+#define _FN1 1
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
```
Comment (fauxpark) [inline keyboards/projectkb/alice/keymaps/default/keymap.c]: I don't see any usages of this keycode, so you can remove this enum if you're not using it.

File: keyboards/projectkb/alice/keymaps/default/keymap.c (hunk 7 above)
Comment (awkannan) [reply keyboards/projectkb/alice/keymaps/default/keymap.c]: lol you're right. I've been copy pasting these from other keyboards for so long I never stopped to really read the code and realize they weren't necessary.

Comment (fauxpark) [reply keyboards/projectkb/alice/keymaps/default/keymap.c]: It's not exactly intuitive considering the `#define LAYOUT(...)`s *do* require the backslashes 😄 And also, it's still pretty prevalent in the repo... and the keymap.c template too.

**Pull request merged**